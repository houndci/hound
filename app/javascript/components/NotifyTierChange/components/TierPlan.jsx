import classNames from 'classnames';
import React from 'react';

export default class TierPlan extends React.Component {
  getIsCurrent() {
    return this.props.isCurrent;
  }

  getIsNew() {
    return this.props.isNew;
  }

  renderCurrentPlan() {
    if (this.getIsCurrent()) {
      return (
        <div className="plan-marker__wrapper">
          <span className="plan-marker plan-marker--current">Current Plan</span>
        </div>
      );
    }
  }

  renderNewPlan() {
    if (this.getIsNew()) {
      return (
        <div className="plan-marker__wrapper">
          <span className="plan-marker plan-marker--new">New Plan</span>
        </div>
      );
    }
  }

  render() {
    const plan = this.props.plan;

    let allowance = null;

    if (plan.allowance > 0) {
      allowance = <div className="plan-allowance">
        Up to <strong>{plan.allowance}</strong> Repos
      </div>;
    } else {
      allowance = <div className="plan-allowance">
        Unlimited
      </div>;
    }

    return(
      <div className={
        classNames(
          "plan",
          "plan-vertical",
          { "current": this.getIsCurrent() },
          { "new": this.getIsNew() }
        )
      }>
        {this.renderCurrentPlan()}
        {this.renderNewPlan()}
        <div className="plan-divider">
          <h5 className="plan-title">{plan.name}</h5>
          {allowance}
        </div>
        <div className="plan-price">${plan.price} <small>month</small></div>
      </div>
    );
  }
}
