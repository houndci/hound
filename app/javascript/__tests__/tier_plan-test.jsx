import TierPlan from '../components/NotifyTierChange/components/TierPlan';

it('renders appropriately (current plan)', () => {
  const plan = {
    name: "Chihuahua",
    price: 49,
    allowance: 4,
    current: true
  };

  const wrapper = shallow(
    <TierPlan
      isCurrent={true}
      isNew={true}
      key={plan.name}
      plan={plan}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

it('renders appropriately (new plan)', () => {
  const plan = {
    name: "Chihuahua",
    price: 49,
    allowance: 4,
    current: false
  };

  const wrapper = shallow(
    <TierPlan
      isCurrent={false}
      isNew={true}
      key={plan.name}
      plan={plan}
    />
  );
  expect(wrapper).toMatchSnapshot();
});

it('renders appropriately (neither current nor new plan)', () => {
  const plan = {
    name: "Chihuahua",
    price: 49,
    allowance: 4,
    current: false
  };

  const wrapper = shallow(
    <TierPlan
      isCurrent={false}
      isNew={false}
      key={plan.name}
      plan={plan}
    />
  );
  expect(wrapper).toMatchSnapshot();
});


it('renders appropriately (bulk)', () => {
  const plan = {
    name: "Bulk",
    price: 0,
    allowance: 0,
    current: true
  };

  const wrapper = shallow(
    <TierPlan
      isCurrent={true}
      isNew={true}
      key={plan.name}
      plan={plan}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
