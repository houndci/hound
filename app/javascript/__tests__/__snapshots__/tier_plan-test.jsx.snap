exports[`test renders appropriately (bulk) 1`] = `
<div
  className="plan plan-vertical current new">
  <div
    className="plan-marker__wrapper">
    <span
      className="plan-marker plan-marker--current">
      Current Plan
    </span>
  </div>
  <div
    className="plan-marker__wrapper">
    <span
      className="plan-marker plan-marker--new">
      New Plan
    </span>
  </div>
  <div
    className="plan-divider">
    <h5
      className="plan-title">
      Bulk
    </h5>
    <div
      className="plan-allowance">
      Unlimited
    </div>
  </div>
  <div
    className="plan-price">
    $
     
    <small>
      month
    </small>
  </div>
</div>
`;

exports[`test renders appropriately (current plan) 1`] = `
<div
  className="plan plan-vertical current new">
  <div
    className="plan-marker__wrapper">
    <span
      className="plan-marker plan-marker--current">
      Current Plan
    </span>
  </div>
  <div
    className="plan-marker__wrapper">
    <span
      className="plan-marker plan-marker--new">
      New Plan
    </span>
  </div>
  <div
    className="plan-divider">
    <h5
      className="plan-title">
      Chihuahua
    </h5>
    <div
      className="plan-allowance">
      Up to 
      <strong>
        4
      </strong>
       Repos
    </div>
  </div>
  <div
    className="plan-price">
    $
    49
     
    <small>
      month
    </small>
  </div>
</div>
`;

exports[`test renders appropriately (neither current nor new plan) 1`] = `
<div
  className="plan plan-vertical">
  <div
    className="plan-divider">
    <h5
      className="plan-title">
      Chihuahua
    </h5>
    <div
      className="plan-allowance">
      Up to 
      <strong>
        4
      </strong>
       Repos
    </div>
  </div>
  <div
    className="plan-price">
    $
    49
     
    <small>
      month
    </small>
  </div>
</div>
`;

exports[`test renders appropriately (new plan) 1`] = `
<div
  className="plan plan-vertical new">
  <div
    className="plan-marker__wrapper">
    <span
      className="plan-marker plan-marker--new">
      New Plan
    </span>
  </div>
  <div
    className="plan-divider">
    <h5
      className="plan-title">
      Chihuahua
    </h5>
    <div
      className="plan-allowance">
      Up to 
      <strong>
        4
      </strong>
       Repos
    </div>
  </div>
  <div
    className="plan-price">
    $
    49
     
    <small>
      month
    </small>
  </div>
</div>
`;
